### PR TITLE
Add entry for e2e test.

### PR DESCRIPTION
**Pull request by yuangu-google**

### CODE REVIEW

File: javatests/arcs/android/e2e/BUILD [hunk 1]
```diff
@@ -0,0 +1,29 @@
+load("//tools/build_defs/android:rules.bzl", "android_binary")
+load("//tools/build_defs/kotlin:rules.bzl", "kt_android_library")
+
+licenses(["notice"])
+
+package(default_visibility = ["//visibility:public"])
+
+kt_android_library(
```
Comment (cromwellian) [inline javatests/arcs/android/e2e/BUILD:8]: I think this should be arcs_kt_android_library because we have global kotlinc ops we pass to kotlinc.

Comment (yuangu-google) [reply javatests/arcs/android/e2e/BUILD:8]: I'm not sure about when to use arcs_kt_android_library vs kt_android_library. When should a developer use arcs_kt_android_library?

Comment (csilvestrini) [reply javatests/arcs/android/e2e/BUILD:8]: Always use arcs_kt_android_library. 

The arcs_* versions of all the build rules include all of the options/customisations we need for our project. If you leave it out you might get tripped up by things like lint errors, missing compiler args, etc.

Comment (yuangu-google) [reply javatests/arcs/android/e2e/BUILD:8]: I just had a discussion with Ray on this topic. IMO "Always use arcs_kt_android_library." is a vague declaration as let's say library A uses part of Arcs so it uses arcs_kt_android_library, then library B depends library A, does library B need to use arcs_kt_android_library? I don't think the answer should be yes.

Comment (csilvestrini) [reply javatests/arcs/android/e2e/BUILD:8]: Sorry I think I misunderstood your original question. Were you referring to developers wishing to use Arcs? Developers using Arcs should feel free to use whatever options they want with kt_android_library. Internally, inside the Arcs project (i.e. this repo), always use arcs_kt_android_library. It contains options which ensure that our code builds with both bazel and blaze.

Comment (cromwellian) [reply javatests/arcs/android/e2e/BUILD:8]: The reason to use arcs_kt_android_library is that if you consume any of Arc's 'inline' functions like toRegistration(), the code will compile as if it is inside of your build target, and some internal G3 lint checks will fail to compile. The arcs_ variants of the build rules supply disable_lint_checks for G3.  These lint checks are buggy (e.g. KotlinReflectNeeded) as the triggering code (x::class) needs no reflection, and Kotlin 1.3.70 obviated the need for this check. Class literals are not reflection use, acceessing declarated fields and methods is.

Comment (yuangu-google) [reply javatests/arcs/android/e2e/BUILD:8]: I see. So it's actually our internal rule, right? I was confused in the beginning as we have arcs_kt_schema/arcs_kt_plan which developer facing rules so I thought this one was for developers to use too.

Shall we rename it to arcs_internal_kt_android_library or other names which are more self-explainable?

**Pull request merged**